http_server.go

```diff
@@ -77,6 +80,16 @@ func NewServer(l *slog.Logger, ar authenticator.Request, lister NamespaceLister)
 	}
 }
 
+func (s *NamespaceListerServer) WithTLS(enableTLS bool) *NamespaceListerServer {
+	s.useTls = enableTLS
+	return s
+}
+
+func (s *NamespaceListerServer) WithTLSOpts(tlsOpts []func(*tls.Config)) *NamespaceListerServer {
```

Comment:
we may simplify the code in main by using a variadic func here

```suggestion
func (s *NamespaceListerServer) WithTLSOpts(tlsOpts ...func(*tls.Config)) *NamespaceListerServer {
```

Reply:
Fixed in f9ea3c6.